fhir-tools/src/main/java/com/ibm/fhir/tools/CodeGenerator.java

```diff
@@ -1132,6 +1134,31 @@ private void generateClass(JsonObject structureDefinition, List<String> paths, C
                 }
             }
 
+            // Handle code/coding/codeableconcept fields with required (or maxValueSet) binding to syntax-based value set
+            for (JsonObject elementDefinition : elementDefinitions) {
+                String basePath = elementDefinition.getJsonObject("base").getString("path");
+                if (elementDefinition.getString("path").equals(basePath)) {
+                    String elementName = getElementName(elementDefinition, path);
+                    String fieldName = getFieldName(elementName);
+                    String fieldType = getFieldType(structureDefinition, elementDefinition, false);
+                    if ("Code".equals(fieldType) || "Coding".equals(fieldType) || "CodeableConcept".equals(fieldType)) {
+                        JsonObject binding = getBinding(elementDefinition);
+                        if (binding != null && binding.containsKey("valueSet") && binding.containsKey("strength")) {
+                            String valueSet = binding.getString("valueSet").split("\\|")[0];
+                            if (!"required".equals(binding.getString("strength")) || hasConcepts(valueSet)) {
+                                valueSet = getMaxValueSet(binding);
+                                valueSet = (valueSet != null) ? valueSet.split("\\|")[0] : valueSet;
```

Comment:
should check that the maxValueSet does not have concepts, and if it does, do not generate the value set syntax check

Reply:
Check added.